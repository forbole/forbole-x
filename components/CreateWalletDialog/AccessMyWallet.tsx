import {
  Button,
  DialogActions,
  DialogContent,
  Typography,
  Box,
  DialogContentText,
  ButtonBase,
} from '@material-ui/core';
import useTranslation from 'next-translate/useTranslation';
import React from 'react';
import useStyles from './styles';
import { ImportStage } from '.';

interface AccessMyWalletProps {
  onConfirm(stage: ImportStage): void;
  onWhatIsMnemonicClick(): void;
}

const AccessMyWallet: React.FC<AccessMyWalletProps> = ({ onConfirm, onWhatIsMnemonicClick }) => {
  const { t } = useTranslation('common');
  const classes = useStyles();

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{t('access my wallet description')}</DialogContentText>
        <Box my={-2}>
          {Object.values(ImportStage).map(stage => (
            <ButtonBase
              key={stage}
              className={classes.borderedButton}
              onClick={() => onConfirm(stage)}>
              <Typography variant="h6" gutterBottom>
                {t(stage)}
              </Typography>
              <Typography variant="body2" color="textSecondary">
                {t(`${stage} description`)}
              </Typography>
            </ButtonBase>
          ))}
        </Box>
      </DialogContent>
      <DialogActions>
        <Button onClick={onWhatIsMnemonicClick} className={classes.button} color="primary">
          {t('what is secret recovery phrase')}
        </Button>
      </DialogActions>
    </>
  );
};

export default AccessMyWallet;
