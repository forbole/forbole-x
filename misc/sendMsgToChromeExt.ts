const sendMsgToChromeExt = (msg: ChromeMessage) =>
  new Promise<any>((resolve, reject) => {
    chrome.runtime.sendMessage(process.env.NEXT_PUBLIC_CHROME_EXT_ID, msg, (response) => {
      if (response.err) {
        console.log(msg, response)
        reject(new Error(response.err))
      } else {
        resolve(response)
      }
    })
  })

export default sendMsgToChromeExt
