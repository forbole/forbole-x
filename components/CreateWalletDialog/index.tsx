import { Dialog, DialogTitle, IconButton } from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import { Cosmos } from 'ledger-app-cosmos'
import CloseIcon from '../../assets/images/icons/icon_cross.svg'
import BackIcon from '../../assets/images/icons/icon_back.svg'
import useStyles from './styles'
import useIconProps from '../../misc/useIconProps'
import CreateWallet from './CreateWallet'
import ConfirmMnemonic from './ConfirmMnemonic'
import { useWalletsContext } from '../../contexts/WalletsContext'
import SecurityPassword from './SecurityPassword'
import ImportWallet from './ImportWallet'
import AccessMyWallet from './AccessMyWallet'
import WhatIsMnemonic from './WhatIsMnemonic'
import Start from './Start'
import ImportMnemonicBackup from './ImportMnemonicBackup'
import sendMsgToChromeExt from '../../misc/sendMsgToChromeExt'
import useStateHistory from '../../misc/useStateHistory'
import ConnectLedgerDialogContent from '../ConnectLedgerDialogContent'
import useIsMobile from '../../misc/useIsMobile'
import { connectLedger } from '../../misc/ledger'

let ledgerApp: Cosmos

export enum ImportStage {
  ImportMnemonicPhraseStage = 'import mnemonic phrase',
  MnemonicPhraseBackupStage = 'use mnemonic phrase backup',
  ConnectLedgerDeviceStage = 'connect ledger device',
}

export enum CommonStage {
  StartStage = 'start',
  AccessMyWalletStage = 'access my wallet',
  CreateWalletStage = 'create wallet',
  ConfirmMnemonicStage = 'confirm mnemonic',
  SetSecurityPasswordStage = 'set security password',
  ImportWalletStage = 'import wallet',
  ImportLedgerWalletStage = 'import ledger wallet',
  WhatIsMnemonicStage = 'what is mnemonic',
}

type Stage = CommonStage | ImportStage

interface CreateWalletDialogProps {
  open: boolean
  onClose(): void
  initialStage?: Stage
}

interface Content {
  title: string
  content: React.ReactNode
}

const CreateWalletDialog: React.FC<CreateWalletDialogProps> = ({ open, onClose, initialStage }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const iconProps = useIconProps()
  const { addWallet } = useWalletsContext()
  const isMobile = useIsMobile()
  const [stage, setStage, toPrevStage, isPrevStageAvailable] = useStateHistory<Stage>(
    initialStage || CommonStage.StartStage
  )
  const [mnemonic, setMnemonic] = React.useState('')
  const [securityPassword, setSecurityPassword] = React.useState('')
  const [error, setError] = React.useState('')

  React.useEffect(() => {
    if (open) {
      setMnemonic('')
      setSecurityPassword('')
      setError('')
      setStage(initialStage || CommonStage.StartStage, true)
    }
  }, [open, initialStage])

  const createWallet = React.useCallback(async () => {
    const wallet = await sendMsgToChromeExt({ event: 'generateMnemonic' })
    setMnemonic(wallet.mnemonic)
    setStage(CommonStage.CreateWalletStage)
  }, [setMnemonic])

  const importMnemonic = React.useCallback(
    async (input) => {
      try {
        const wallet = await sendMsgToChromeExt({
          event: 'verifyMnemonic',
          data: { mnemonic: input },
        })
        setMnemonic(wallet.mnemonic)
        setStage(CommonStage.SetSecurityPasswordStage)
      } catch (err) {
        setError(t(err.message))
      }
    },
    [setMnemonic, setStage, setError]
  )

  const importMnemonicBackup = React.useCallback(
    async (data) => {
      try {
        const wallet = await sendMsgToChromeExt({
          event: 'verifyMnemonicBackup',
          data,
        })
        setMnemonic(wallet.mnemonic)
        setStage(CommonStage.SetSecurityPasswordStage)
      } catch (err) {
        setError(t(err.message))
      }
    },
    [setMnemonic, setStage, setError]
  )

  const confirmMnemonic = React.useCallback(
    (input) => {
      if (input === mnemonic) {
        setStage(CommonStage.SetSecurityPasswordStage)
      } else {
        setError(t('invalid mnemonic'))
      }
    },
    [mnemonic, setStage, setError]
  )

  const confirmSecurityPassword = React.useCallback(
    (pw: string) => {
      setSecurityPassword(pw)
      setStage(CommonStage.ImportWalletStage)
    },
    [setStage, setSecurityPassword]
  )

  const saveWallet = React.useCallback(
    async (name: string, cryptos: string[], type = 'mnemonic') => {
      await addWallet(
        {
          type,
          name,
          cryptos,
          mnemonic,
          securityPassword,
        },
        ledgerApp
      )
      onClose()
    },
    [addWallet, mnemonic, securityPassword, onClose]
  )

  const content: Content = React.useMemo(() => {
    switch (stage) {
      case ImportStage.ConnectLedgerDeviceStage:
        return {
          title: '',
          content: (
            <ConnectLedgerDialogContent
              onConnect={async () => {
                ledgerApp = await connectLedger()
                setStage(CommonStage.ImportLedgerWalletStage, undefined, true)
              }}
            />
          ),
        }
      case ImportStage.ImportMnemonicPhraseStage:
        return {
          title: t('mnemonic'),
          content: (
            <ConfirmMnemonic
              description={t('mnemonic description')}
              onConfirm={importMnemonic}
              error={error}
            />
          ),
        }
      case ImportStage.MnemonicPhraseBackupStage:
        return {
          title: t('mnemonic phrase backup title'),
          content: <ImportMnemonicBackup onConfirm={importMnemonicBackup} error={error} />,
        }
      case CommonStage.CreateWalletStage:
        return {
          title: t('create new wallet title'),
          content: (
            <CreateWallet
              mnemonic={mnemonic}
              onConfirm={() => setStage(CommonStage.ConfirmMnemonicStage)}
            />
          ),
        }
      case CommonStage.ConfirmMnemonicStage:
        return {
          title: t('create new wallet title'),
          content: (
            <ConfirmMnemonic
              description={t('confirm mnemonic description')}
              onConfirm={confirmMnemonic}
              error={error}
            />
          ),
        }
      case CommonStage.SetSecurityPasswordStage:
        return {
          title: t('security password title'),
          content: <SecurityPassword onConfirm={confirmSecurityPassword} />,
        }
      case CommonStage.ImportWalletStage:
      case CommonStage.ImportLedgerWalletStage:
        return {
          title: t('import wallet title'),
          content: (
            <ImportWallet
              onConfirm={(name, cryptos) =>
                saveWallet(
                  name,
                  cryptos,
                  stage === CommonStage.ImportLedgerWalletStage ? 'ledger' : 'mnemonic'
                )
              }
            />
          ),
        }
      case CommonStage.WhatIsMnemonicStage:
        return {
          title: t('what is mnemonic phrase'),
          content: <WhatIsMnemonic />,
        }
      case CommonStage.AccessMyWalletStage:
        return {
          title: t('access my wallet title'),
          content: <AccessMyWallet onConfirm={setStage} onCreateWallet={createWallet} />,
        }
      case CommonStage.StartStage:
      default:
        return {
          title: t('create wallet title'),
          content: (
            <Start
              onWhatIsMnemonicClick={() => setStage(CommonStage.WhatIsMnemonicStage)}
              onCreateWalletClick={createWallet}
              onImportWalletClick={() => setStage(CommonStage.AccessMyWalletStage)}
            />
          ),
        }
    }
  }, [stage, t])

  return (
    <Dialog
      fullWidth
      open={open}
      onClose={onClose}
      fullScreen={isMobile}
      PaperProps={{
        className: classes.dialog,
      }}
    >
      {isPrevStageAvailable ? (
        <IconButton className={classes.backButton} onClick={toPrevStage}>
          <BackIcon {...iconProps} />
        </IconButton>
      ) : null}
      <IconButton className={classes.closeButton} onClick={onClose}>
        <CloseIcon {...iconProps} />
      </IconButton>
      <DialogTitle>{content.title}</DialogTitle>
      {content.content}
    </Dialog>
  )
}

export default CreateWalletDialog
