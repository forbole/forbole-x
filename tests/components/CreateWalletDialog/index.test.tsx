import React from 'react'
import renderer from 'react-test-renderer'
import CreateWalletDialog, {
  ImportStage,
  CommonStage,
} from '../../../components/CreateWalletDialog'
import sendMsgToChromeExt from '../../../misc/sendMsgToChromeExt'

const mockWalletsContext = {
  addWallet: jest.fn(),
}

const onClose = jest.fn()

jest.mock('../../../contexts/WalletsContext', () => ({
  useWalletsContext: () => mockWalletsContext,
}))

jest.mock('../../../misc/sendMsgToChromeExt', () => jest.fn())

jest.mock('../../../components/CreateWalletDialog/CreateWallet', () => (props) => (
  <div id="CreateWallet" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/ConfirmMnemonic', () => (props) => (
  <div id="ConfirmMnemonic" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/SecurityPassword', () => (props) => (
  <div id="SecurityPassword" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/ImportWallet', () => (props) => (
  <div id="ImportWallet" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/AccessMyWallet', () => (props) => (
  <div id="AccessMyWallet" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/ImportMnemonicBackup', () => (props) => (
  <div id="ImportMnemonicBackup" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/WhatIsMnemonic', () => (props) => (
  <div id="WhatIsMnemonic" {...props} />
))
jest.mock('@material-ui/core/Dialog', () => (props) => <div id="dialog" {...props} />)

describe('component: CreateWalletDialog', () => {
  it('renders open state correctly', () => {
    const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders closed state correctly', () => {
    const component = renderer.create(<CreateWalletDialog open={false} onClose={onClose} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  // it('renders import wallets stage correctly', () => {
  //   const component = renderer.create(<CreateWalletDialog open={false} onClose={onClose} />)
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onConfirm(ImportStage)
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('renders correctly when back button is clicked on import wallets stage', () => {
  //   const component = renderer.create(<CreateWalletDialog open={false} onClose={onClose} />)
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onConfirm(ImportStage)
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         'data-file-name': 'SvgIcon_back',
  //       })
  //       .parent.parent.parent.props.onClick()
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('calls onClose when close icon button is clicked', () => {
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         'data-file-name': 'SvgIcon_cross',
  //       })
  //       .parent.parent.parent.props.onClick()
  //   })
  //   expect(onClose).toBeCalled()
  // })
  // it('calls Secp256k1HdWallet.generate(24) and renders create wallet stage correctly when onCreateWalletClick is called', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockResolvedValueOnce({ mnemonic: 'mnemonic' })
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onCreateWallet()
  //   })
  //   expect(sendMsgToChromeExt).toBeCalledWith({ event: 'generateMnemonic' })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('renders confirm mnemonic stage correctly when onConfirm is called from CreateWallet', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockResolvedValueOnce({ mnemonic: 'mnemonic' })
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onCreateWallet()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'CreateWallet',
  //       })
  //       .props.onConfirm()
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('renders set security password stage correctly when mnemonic is confirmed to be correct', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockResolvedValueOnce({ mnemonic: 'mnemonic' })
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onCreateWallet()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'CreateWallet',
  //       })
  //       .props.onConfirm()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'ConfirmMnemonic',
  //       })
  //       .props.onConfirm('mnemonic')
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('renders correctly when confirm mnemonic is invalid', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockResolvedValueOnce({ mnemonic: 'mnemonic' })
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onCreateWallet()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'CreateWallet',
  //       })
  //       .props.onConfirm()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'ConfirmMnemonic',
  //       })
  //       .props.onConfirm('invalid mnemonic')
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('renders import wallet stage correctly when security password is set', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockResolvedValueOnce({ mnemonic: 'mnemonic' })
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onCreateWallet()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'CreateWallet',
  //       })
  //       .props.onConfirm()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'ConfirmMnemonic',
  //       })
  //       .props.onConfirm('mnemonic')
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'SecurityPassword',
  //       })
  //       .props.onConfirm('123123')
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('calls addWallet and onClose when wallet name is set', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockResolvedValueOnce({ mnemonic: 'mnemonic' })
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onCreateWallet()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'CreateWallet',
  //       })
  //       .props.onConfirm()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'ConfirmMnemonic',
  //       })
  //       .props.onConfirm('mnemonic')
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'SecurityPassword',
  //       })
  //       .props.onConfirm('123123')
  //   })
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'ImportWallet',
  //       })
  //       .props.onConfirm('wallet name', ['DSM'])
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  //   expect(mockWalletsContext.addWallet).toBeCalledWith(
  //     {
  //       mnemonic: 'mnemonic',
  //       name: 'wallet name',
  //       securityPassword: '123123',
  //       type: 'mnemonic',
  //       cryptos: ['DSM'],
  //     },
  //     undefined
  //   )
  //   expect(onClose).toBeCalled()
  // })
  // it('renders import mnemonic phrase stage correctly when onConfirm is called from AccessMyWallet', async () => {
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onConfirm(ImportStage.ImportMnemonicPhraseStage)
  //   })
  //   // renderer.act(() => {
  //   //   component.root
  //   //     .findByProps({
  //   //       id: 'AccessMyWallet',
  //   //     })
  //   //     .props.onConfirm(ImportStage.ImportMnemonicPhraseStage)
  //   // })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('calls verify mnemonic and renders set security password stage correctly when onConfirm is called from ConfirmMnemonic', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockResolvedValueOnce({ mnemonic: 'mnemonic' })
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onConfirm(ImportStage.ImportMnemonicPhraseStage)
  //   })
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'ConfirmMnemonic',
  //       })
  //       .props.onConfirm('mnemonic')
  //   })
  //   expect(sendMsgToChromeExt).toBeCalledWith({
  //     event: 'verifyMnemonic',
  //     data: { mnemonic: 'mnemonic' },
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('renders error state when invalid mnemonic phrase is imported', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockRejectedValueOnce(new Error('invalid mnemonic'))
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onConfirm(ImportStage.ImportMnemonicPhraseStage)
  //   })
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'ConfirmMnemonic',
  //       })
  //       .props.onConfirm('mnemonic')
  //   })
  //   expect(sendMsgToChromeExt).toBeCalledWith({
  //     event: 'verifyMnemonic',
  //     data: { mnemonic: 'mnemonic' },
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('renders mnemonic phrase backup stage correctly when onConfirm is called from AccessMyWallet', async () => {
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onConfirm(ImportStage.MnemonicPhraseBackupStage)
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('calls verify mnemonic backup and renders set security password stage correctly when onConfirm is called from ImportMnemonicBackup', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockResolvedValueOnce({ mnemonic: 'mnemonic' })
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onConfirm()
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onConfirm(ImportStage.MnemonicPhraseBackupStage)
  //   })
  //   await renderer.act(async () => {
  //     await component.root
  //       .findByProps({
  //         id: 'ImportMnemonicBackup',
  //       })
  //       .props.onConfirm({
  //         password: 'password',
  //         backupPhrase: 'backup phrase',
  //       })
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
  // it('renders error state when invalid mnemonic phrase backup is imported', async () => {
  //   ;(sendMsgToChromeExt as jest.Mock).mockRejectedValueOnce(new Error('invalid mnemonic'))
  //   const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
  //   await renderer.act(async () => {
  //     component.root
  //       .findByProps({
  //         id: 'AccessMyWallet',
  //       })
  //       .props.onConfirm(ImportStage.MnemonicPhraseBackupStage)
  //   })
  //   renderer.act(() => {
  //     component.root
  //       .findByProps({
  //         id: 'ImportMnemonicBackup',
  //       })
  //       .props.onConfirm({
  //         password: 'password',
  //         backupPhrase: 'backup phrase',
  //       })
  //   })
  //   const tree = component.toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
})

afterEach(() => {
  jest.clearAllMocks()
})
