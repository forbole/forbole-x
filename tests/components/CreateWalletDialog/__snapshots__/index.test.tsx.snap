// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CreateWalletDialog renders closed state correctly 1`] = `
<div
  PaperProps={
    Object {
      "className": "HookGlobalStyles-dialog-8",
    }
  }
  fullScreen={false}
  fullWidth={true}
  id="dialog"
  onClose={[MockFunction]}
  open={false}
>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-closeButton-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_cross"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:create wallet title
    </h2>
  </div>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:create wallet description
    </p>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <button
          className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiBox-root MuiBox-root-11"
          >
            <svg
              data-file-name="SvgLogin_light"
            />
          </div>
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            common:i have mnemonic phrase
          </p>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <button
          className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiBox-root MuiBox-root-12"
          >
            <svg
              data-file-name="SvgCreate_wallet_light"
            />
          </div>
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            common:i dont have mnemonic phrase
          </p>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text HookGlobalStyles-button-1 MuiButton-textPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:what is mnemonic phrase
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`component: CreateWalletDialog renders open state correctly 1`] = `
<div
  PaperProps={
    Object {
      "className": "HookGlobalStyles-dialog-8",
    }
  }
  fullScreen={false}
  fullWidth={true}
  id="dialog"
  onClose={[MockFunction]}
  open={true}
>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-closeButton-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_cross"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
  </button>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:create wallet title
    </h2>
  </div>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:create wallet description
    </p>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <button
          className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiBox-root MuiBox-root-9"
          >
            <svg
              data-file-name="SvgLogin_light"
            />
          </div>
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            common:i have mnemonic phrase
          </p>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <button
          className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiBox-root MuiBox-root-10"
          >
            <svg
              data-file-name="SvgCreate_wallet_light"
            />
          </div>
          <p
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            common:i dont have mnemonic phrase
          </p>
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text HookGlobalStyles-button-1 MuiButton-textPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        common:what is mnemonic phrase
      </span>
    </button>
  </div>
</div>
`;
